### DISCUSSION TITLE
Is it possible to run everything with SSL?

### DISCUSSION THREAD

**Discussion by jhill2727** [seaweedfs/seaweedfs#4079] [General]
I've got a seaweedfs docker setup that will spin up with the volume server using https and nginx proxies in front of filer and master using https. Everything works except actually writing the file. We can programmatically create new buckets and create the file, but when writing it, we get errors like:

E1221 18:17:41     1 upload_content.go:293] unmarshal http://172.17.2.177:8888/?proxyChunkId=31,1a0e2ab727: Client sent an HTTP request to an HTTPS server.
W1221 18:17:41     1 upload_content.go:114] uploading to http://172.17.2.177:8888/?proxyChunkId=31,1a0e2ab727: unmarshal http://172.17.2.177:8888/?proxyChunkId=31,1a0e2ab727: invalid character 'C' looking for beginning of value
E1221 18:17:41     1 upload_content.go:293] unmarshal http://172.17.2.177:8888/?proxyChunkId=31,1a0e2ab727: Client sent an HTTP request to an HTTPS server.
W1221 18:17:41     1 upload_content.go:114] uploading to http://172.17.2.177:8888/?proxyChunkId=31,1a0e2ab727: unmarshal http://172.17.2.177:8888/?proxyChunkId=31,1a0e2ab727: invalid character 'C' looking for beginning of value

Is there any way to make that request https? If we go back to http settings, obviously all is well. But that causes issues with mixed content in our application.

**Comment by chrislusf** [comment]
there are many places using "http://" directly.
